packages/calypso-e2e/src/lib/pages/gutenberg-editor-page.ts

```diff
@@ -143,6 +154,60 @@ export class GutenbergEditorPage extends BaseContainer {
 		return lines.filter( Boolean ).join( '\n' );
 	}
 
+	/**
+	 * Given a name, adds the Gutenberg block matching the name.
+	 *
+	 * The name is expected to be formatted in the same manner as it
+	 * appears on the label when visible in the block inserter panel.
+	 *
+	 * Example:
+	 * 		- Click to Tweet
+	 * 		- Pay with Paypal
+	 * 		- SyntaxHighlighter Code
+	 *
+	 * @param {string} blockName Name of the block to be inserted.
+	 */
+	async addBlock( blockName: string ): Promise< ElementHandle > {
+		// Click on the editor title. This has the effect of dismissing the block inserter
+		// if open, and restores focus back to the editor root container, allowing insertion
+		// of blocks.
+		await this.frame.click( selectors.editorTitle );
+		await this.openBlockInserter();
+		await this.frame.fill( selectors.blockSearch, blockName );
+		await this.frame.click( `${ selectors.blockInserterResultItem }:has-text("${ blockName }")` );
+		// Confirm the block has been added to the editor body.
+		return await this.frame.waitForSelector( `[aria-label="Block: ${ blockName }"]` );
+	}
```

Comment:
I just wanted to say I really like this pattern you've locked in where the `GutenbergEditorPage` serves almost as a factory to create the ElementHandles we then feed as dependencies for Block POMs. It reads really well and I think it's going to grow and scale fantastically!

Reply:
Hey look, unintentional factory pattern 😆

Reply:
This is a great improvement over the existing `addBlock` method, I like how concise is!